react-form-controls/src/index.jsx

```diff
@@ -0,0 +1,45 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+
+class RegistrationForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { username: '', password: '' };
+    this.handleUsernameChange = this.handleUsernameChange.bind(this);
+    this.handlePasswordChange = this.handlePasswordChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleUsernameChange(event) {
+    this.setState({ username: event.target.value });
+  }
+
+  handlePasswordChange(event) {
+    this.setState({ password: event.target.value });
+  }
+
+  handleSubmit(event) {
+    console.log('state:', this.state);
+    event.preventDefault();
+  }
+
+  render() {
+    return (
+    <div>
+      <label>
+          Username
+          <input type="text" value={this.state.username} onChange={this.handleUsernameChange} />
+      </label>
+      <label>
+        Password
+          <input type="password" value={this.state.password} onChange={this.handlePasswordChange}/>
+      </label>
+      <button onClick={this.handleSubmit}>Sign Up</button>
```

Comment:
Instead of listening for clicks on button elements for form submission, we listen for submits on form elements

```jsx
<form onSubmit={this.handleSubmit}>
```

The button (and the Enter key) will trigger the submit on the `<form>` element.